apps/web/src/views/BuyCrypto/hooks/usePriceQuoter.ts

```diff
@@ -68,49 +67,61 @@ const usePriceQuotes = (amount: string, inputCurrency: string, outputCurrency: s
   const [quotes, setQuotes] = useState<ProviderQoute[]>([])
   const { chainId } = useActiveChainId()
 
-  const fetchProviderAvailability = async (ip: string, combinedData: ProviderQoute[]) => {
-    // first check user availability
-    const responsePromises = [fetchMoonpayAvailability(ip), fetchMercuryoAvailability(ip)]
-    const responses = await Promise.allSettled(responsePromises)
+  const fetchProviderAvailability = useCallback(
+    async (ip: string, combinedData: ProviderQoute[]) => {
+      // first check user availability
+      const responsePromises = [fetchMoonpayAvailability(ip), fetchMercuryoAvailability(ip)]
+      const responses = await Promise.allSettled(responsePromises)
```

Comment:
Always a good practice to not define unnecessary variable. We can remove the `responsePromises`